server/src/main/java/org/elasticsearch/action/synonyms/DeleteSynonymRuleAction.java

```diff
@@ -0,0 +1,149 @@
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.support.master.AcknowledgedResponse.ACKNOWLEDGED_KEY;
+
+public class DeleteSynonymRuleAction extends ActionType<DeleteSynonymRuleAction.Response> {
+
+    public static final DeleteSynonymRuleAction INSTANCE = new DeleteSynonymRuleAction();
+    public static final String NAME = "cluster:admin/synonym_rules/delete";
+
+    public DeleteSynonymRuleAction() {
+        super(NAME, DeleteSynonymRuleAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+        private final String synonymsSetId;
+
+        private final String synonymRuleId;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.synonymsSetId = in.readString();
+            this.synonymRuleId = in.readString();
+        }
+
+        public Request(String synonymsSetId, String synonymRuleId) {
+            this.synonymsSetId = synonymsSetId;
+            this.synonymRuleId = synonymRuleId;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+            if (Strings.isNullOrEmpty(synonymsSetId)) {
+                validationException = ValidateActions.addValidationError("synonyms set must be specified", validationException);
+            }
+            if (Strings.isNullOrEmpty(synonymRuleId)) {
+                validationException = ValidateActions.addValidationError("synonym rule id must be specified", validationException);
+            }
+
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(synonymsSetId);
+            out.writeString(synonymRuleId);
+        }
+
+        public String synonymsSetId() {
+            return synonymsSetId;
+        }
+
+        public String synonymRuleId() {
+            return synonymRuleId;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return Objects.equals(synonymsSetId, request.synonymsSetId) && Objects.equals(synonymRuleId, request.synonymRuleId);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(synonymsSetId, synonymRuleId);
+        }
+    }
+
+    // TODO Remove duplicates
```

Comment:
I know you said there would be a followup PR, but should we merge this with the `TODO` line? It feels like we might want to take it out before merging.

Reply:
I think there are TODOs in other parts of the codebase. I hope I get this removed soon enough ;)